apps/gdalalg_vector_pipeline.h

```diff
@@ -219,6 +221,76 @@ class GDALVectorPipelinePassthroughLayer /* non final */
     }
 };
 
+/**
+ * Dataset used to read all input features into memory and perform some
+ * processing.
+ */
+class GDALVectorNonStreamingAlgorithmDataset : public GDALDataset
+{
+  public:
+    GDALVectorNonStreamingAlgorithmDataset()
+    {
+        GDALDriver *poMemDriver =
+            GetGDALDriverManager()->GetDriverByName("MEM");
+        m_ds.reset(cpl::down_cast<MEMDataset *>(
+            poMemDriver->Create("", 0, 0, 0, GDT_Unknown, nullptr)));
+    }
+
+    virtual bool Process(OGRLayer &srcLayer, OGRLayer &dstLayer) = 0;
+
+    bool AddProcessedLayer(OGRLayer &srcLayer)
+    {
+        OGRMemLayer *poDstLayer = cpl::down_cast<OGRMemLayer *>(
```

Comment:
We could test srcLayer.TestCapability(OLCStringsAsUTF8), and if set pass the ADVERTIZE_UTF8=YES layer creation option

Reply:
Done in 20d580e62ed3d90a3655e305f13a8c8d53f92060